bika/lims/browser/sample/view.py

```diff
@@ -417,46 +430,46 @@ def folderitem(self, obj, item, index):
                  self.portal_url)
         if after_icons:
             item['after']['getSampleID'] = after_icons
-
-        if obj.getSamplingWorkflowEnabled():
+        if obj.getSamplingWorkflowEnabled:
             datesampled = self.ulocalized_time(
-                obj.getDateSampled(), long_format=True)
+                obj.getDateSampled, long_format=True)
             if not datesampled:
                 datesampled = self.ulocalized_time(
                     DateTime(), long_format=True)
                 item['class']['DateSampled'] = 'provisional'
-            sampler = obj.getSampler().strip()
+            sampler = obj.getSampler.strip()
             if sampler:
                 item['replace']['getSampler'] = self.user_fullname(sampler)
             if 'Sampler' in member.getRoles() and not sampler:
                 sampler = member.id
                 item['class']['getSampler'] = 'provisional'
         else:
-            datesampled = self.ulocalized_time(obj.getDateSampled(), long_format=True)
+            datesampled = self.ulocalized_time(obj.getDateSampled, long_format=True)
             sampler = ''
 
         item['DateSampled'] = datesampled
         item['getSampler'] = sampler
         # sampling workflow - inline edits for Sampler, Date Sampled and
         # Scheduled Sampling Sampler
         checkPermission = self.context.portal_membership.checkPermission
-        state = workflow.getInfoFor(obj, 'review_state')
+        state = obj.review_state
         if state in ['to_be_sampled', 'scheduled_sampling']:
             item['required'] = []
             item['allow_edit'] = []
             item['choices'] = {}
-            samplers = getUsers(obj, ['Sampler', 'LabManager', 'Manager'])
+            samplers = getUsers(self.context, ['Sampler', 'LabManager', 'Manager'])
```

Comment:
I suggest to retrieve samplers outside the `folderitem` function (maybe in `__init__` or at the beginning of `folderitems` - note the plural form - functions). Note that `folderitem` (singular form) function is called each time an item has to be rendered, so `getUsers` is called as many times as number of items to be displayed.

Reply:
Indeed, you can use the "beforerender" method may be.